<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GameWindow</class>
 <widget class="QMainWindow" name="GameWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>969</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GameWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0" rowspan="3">
     <widget class="QGroupBox" name="gb_grid">
      <property name="title">
       <string>Grid</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>width:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QSpinBox" name="sb_grid_width">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>5</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>40</number>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>height:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QSpinBox" name="sb_grid_height">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>5</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>30</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>fill factor:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2" colspan="2">
        <widget class="QSpinBox" name="sb_fill_factor">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="suffix">
          <string>%</string>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="value">
          <number>30</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="5">
        <widget class="QPushButton" name="bt_generate_biome">
         <property name="text">
          <string>Generate random biome</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="gb_rules">
      <property name="title">
       <string>Rules</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QRadioButton" name="rb_rules_conway">
         <property name="text">
          <string>Conway's Game of Life</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rb_rules_life_like">
         <property name="text">
          <string>Life-like</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cb_life_like_rules">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Notation:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="le_rule_notation">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="gb_colors">
      <property name="title">
       <string>Colors</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QRadioButton" name="rb_color_black_white">
         <property name="text">
          <string>black/white</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rb_color_immigration">
         <property name="text">
          <string>Immigration</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rb_color_quadlife">
         <property name="text">
          <string>QuadLife</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QGroupBox" name="gb_simulation">
      <property name="title">
       <string>Simulation</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="bt_single_step">
         <property name="text">
          <string>Single step</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bt_sim_start_stop">
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Time step:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sb_time_step">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="suffix">
          <string> ms</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>2000</number>
         </property>
         <property name="singleStep">
          <number>100</number>
         </property>
         <property name="value">
          <number>200</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Jump:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sb_jump">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>999999</number>
         </property>
         <property name="singleStep">
          <number>100</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bt_jump">
         <property name="text">
          <string>Do Jump</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QGraphicsView" name="game_view"/>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="status_bar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
